### ISSUE TITLE
EnvironmentVariable operations with EnvironmentVariableTarget.User fails on Windows Nano Server

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/corefx#40226]
This seems to be a product bug because of an underlying windows api change.

cc @Anipik @stephentoub @danmosemsft

**Comment by ViktorHofer** [comment]
Setting to 3.0 to make sure that the issue is triaged asap.

**Comment by ViktorHofer** [comment]
Disabled the test in https://github.com/dotnet/corefx/pull/39734/commits/a96948828a326250c7be84dc2aabb5ed40caea55.

**Comment by Anipik** [comment]
@ViktorHofer is any exception being thrown here ? do you have the failure\exception message ?

**Comment by ViktorHofer** [comment]
No exception being thrown in source code. Just assert errors in the different tests that test the behavior.

**Comment by danmosemsft** [comment]
This may have to move out of 3.0 now given there's no customer reports (?)

@ViktorHofer was it previously passing on a different version of Nano, and the upgrade broke it?

**Comment by ViktorHofer** [comment]
> was it previously passing on a different version of Nano, and the upgrade broke it?

Yes, the upgrade to Windows Nano Server 1809 broke this. I assume the number of customers that use .NET Core on Nano is limited.

**Comment by ViktorHofer** [comment]
We should look at the api changes in Nano 1809 and/or talk with the Windows team to understand the issue better. This isn't 3.0 specific so I'm fine moving it out but it probably affects all .NET Core versions that support Nano and use that API.

**Comment by maryamariyan** [comment]
I ran the test on Win10, Windows Nano Server 1809 and 1803.

The repro (test failure: SetEnvironmentVariable failure) only happened in 1809.
The reason is ```HKEY_CURRENT_USER\Environment``` registry was not present.
After manually adding that in 1809, the test passes in 1809 as well.

**Comment by danmosemsft** [comment]
Sounds like we could reasonably work around that in code then (eg create hte key or skip the operation or whatever)

**Comment by ericstj** [comment]
Skipping the operation is what happens now: that's the bug.  We should probably let the Windows folks know about this regression and confirm that removal of the key was intentional.

We could create the key, assuming we can get the ACLs right.  On my machine it looks like this key has an explicit ACL (not inherited).  We'd want to test what would happen when we create the key from elevated and non-elevated process (if such a thing exists in nano) to ensure we create the key with appropriate ACL.

We should test if we add the key, and create a new process, that the variable will be set in that process.  Given Nano removed creation of the key, we need to confirm they didn't remove consumption of the key too.

**Comment by ViktorHofer** [comment]
Creating the key ourselves is a bit scary, let's first talk with Windows.
